src/commands/apps-sdkconfig.ts

```diff
@@ -0,0 +1,128 @@
+import * as requireAuth from "../requireAuth";
+import * as logger from "../logger";
+import { promptOnce } from "../prompt";
+
+async function selectAppInteractively(
+  projectId: string,
+  appPlatform: AppPlatform
+): Promise<AppMetadata> {
+  if (!projectId) {
+    throw new FirebaseError("Project ID must not be empty.");
+  }
+
+  const apps: AppMetadata[] = await listFirebaseApps(projectId, appPlatform);
+  if (apps.length === 0) {
+    throw new FirebaseError(
+      `There are no ${appPlatform === AppPlatform.ANY ? "" : appPlatform + " "}apps ` +
+        "associated with this Firebase project"
+    );
+  }
+
+  const choices = apps.map((app: any) => {
+    return {
+      name:
+        `${app.displayName || app.bundleId || app.packageName}` +
+        ` - ${app.appId} (${app.platform})`,
+      value: app,
+    };
+  });
+
+  return await promptOnce({
+    type: "list",
+    name: "id",
+    message:
+      `Select the ${appPlatform === AppPlatform.ANY ? "" : appPlatform + " "}` +
+      "app to get the configuration data:",
+    choices,
+  });
+}
+
+module.exports = new Command("apps:sdkconfig [platform] [appId]")
+  .description(
+    "print the Google Services config of a Firebase app. " +
+      "[platform] can be IOS, ANDROID or WEB (case insensitive)"
+  )
+  .option("-o, --out [file]", "(optional) write config output to a file")
+  .before(requireAuth)
+  .action(
+    async (platform: string = "", appId: string = "", options: any): Promise<any> => {
+      let appPlatform = getAppPlatform(platform);
+
+      if (appPlatform === AppPlatform.PLATFORM_UNSPECIFIED) {
+        throw new FirebaseError("Unexpected platform. Only support iOS, Android and Web apps.");
+      }
+
+      if (!appId) {
+        if (options.nonInteractive) {
+          throw new FirebaseError("App ID must not be empty.");
+        }
+
+        if (!options.project) {
+          // TODO(caot): select project from a list
+          options.project = await promptOnce({
+            type: "input",
+            message: "Please input the project ID you would like to use:",
+          });
+        }
+
+        const appMetadata: AppMetadata = await selectAppInteractively(options.project, appPlatform);
+        appId = appMetadata.appId;
+        appPlatform = appMetadata.platform;
+      }
+
+      let configData;
+      const spinner = ora(
+        `Downloading configuration data of your Firebase ${appPlatform} app`
+      ).start();
+      try {
+        configData = await getAppConfig(appId, appPlatform);
+      } catch (err) {
+        spinner.fail();
+        throw err;
+      }
+
+      spinner.succeed();
+      logger.info("");
+      logger.info(`=== Your app configuration is ready ===`);
+      logger.info("");
+
+      if (options.out === undefined) {
+        logger.info(configData.fileContents);
+        return configData;
+      }
+
+      const fileName =
+        options.out === true || options.out === "" ? configData.fileName : options.out;
```

Comment:
put parenthesis around the logical parts that go together. I'm 99% this gets executed as such (after playing with it a bit myself):

```typescript
(options.out === true || options.out === "") ? configData.fileName : options.out
```

Add the parenthesis to guarantee what is happening

Reply:
Thanks, updated

Reply:
+1 to Bryan's suggestion. If tslint is complaining, then we should either change the tslint config to allow it, or separate the logic like this:

```ts
const shouldUseDefaultFilename = (options.out === true || options.out === "");
const filename = shouldUseDefaultFilename ? configData.fileName : options.out;
```

(btw, `filename` is one word, so the `n` doesn't have to be capitalized)

Reply:
+1 for Nick's suggestion. updated code